lib/cortex/file_watcher.ex

```diff
@@ -26,14 +31,22 @@ defmodule Cortex.FileWatcher do
 
     FileSystem.subscribe(watcher_pid)
 
-    {:ok, %{watcher_pid: watcher_pid}}
+    {:ok, %State{watcher_pid: watcher_pid, throttle_timer: nil, file_events: %{}}}
   end
 
   def handle_info(
         {:file_event, watcher_pid, {path, _events}},
         %{watcher_pid: watcher_pid} = state
       ) do
-    GenServer.cast(Controller, {:file_changed, file_type(path), path})
+    %State{file_events: file_events, throttle_timer: throttle_timer} = state
+
+    throttle_timer =
+      unless throttle_timer do
```

Comment:
won't this set the throttle timer to `nil` the second time the file event is run?

Reply:
Once again you are completely correct. Sorry for the back and forth but I have updated the code to fix this issue.

To help prevent us (well me) from making issues like this in the future it would be nice to add some tests. As part of this change I did write a test, but in my opinion it is too tied to the current implementation (liberal use of `:sys.get_state`) to be a worthwhile test going forward. In order to better test the module it is necessary to change the implementation slightly to facilitate testing, specifically I think the `start_link` function could take an optional argument that would store both the file_changed recipient (defaulting to `Cortex.Controller`) and the throttle_timeout (defaulting to `100`). I'll just put that up as a separate PR.